### PR TITLE
Address audit issues

### PR DESCRIPTION
**Pull request by thal0x**
PR to address findings found in our audit, the document can be found in our Telegram chat with them.

### CODE REVIEW

File: solidity/src/FastTransferGateway.sol
Comment (dhfang) [inline solidity/src/FastTransferGateway.sol:268]: It would also be nice to change _settleOrder to tally the amount to send and modify order status in the same loop like in the cosmwasm side for symmetry and to prevent the duplicate orderID issue at message receive independently of the message send implementation. Will leave it up to your discretion.

**Review by dhfang — APPROVED** [review]
Looks good to me! Thank you for holding off on forwarding to Zellic, just wanted to check that the assumptions the solver makes about semantics are unchanged.

File: solidity/src/FastTransferGateway.sol
Comment (thal0x) [reply solidity/src/FastTransferGateway.sol:268]: Yeah you're right, I updated this

**Pull request merged**
